### PR TITLE
[SYCLomatic] Migrate CU_LAUNCH_PARAM_* macros

### PR DESCRIPTION
**Pull request by LU-JOHN**
Signed-off-by: Lu, John [john.lu@intel.com](mailto:john.lu@intel.com)

### CODE REVIEW

**Review by jzc — APPROVED** [review]

**Review by AndyCHHuang — APPROVED** [review]
LGTM

File: clang/lib/DPCT/MapNames.cpp [hunk 1]
```diff
@@ -4122,6 +4122,19 @@ std::unordered_map<std::string, MacroMigrationRule> MapNames::MacroRuleMap{
     {"cudaStreamDefault",
      MacroMigrationRule("cudaStreamDefault_rule", RulePriority::Fallback,
                         "cudaStreamDefault", "0")},
+
+    {"CU_LAUNCH_PARAM_BUFFER_SIZE",
+     MacroMigrationRule("kernel_param_rule", RulePriority::Fallback,
+                        "CU_LAUNCH_PARAM_BUFFER_SIZE", "DPCT_KERNEL_PARAM_SIZE",
```
Comment (zhimingwang36) [inline clang/lib/DPCT/MapNames.cpp]: Could u migrate those macros to literal values (like 0,1,2)?

Comment (LU-JOHN) [reply clang/lib/DPCT/MapNames.cpp]: Isn't it preferable to use a symbolic value?   If were writing SYCL code to use this interface we wouldn't want them to use magic numbers in their source.

Comment (zhimingwang36) [reply clang/lib/DPCT/MapNames.cpp]: It's quite API specific things, it's not common. From SYCL view, we don't want to introduce this kind new macros in dpct:: namespace.

Comment (LU-JOHN) [reply clang/lib/DPCT/MapNames.cpp]: Updated code to use literal values

**Review by zhimingwang36 — APPROVED** [review]
LGTM

**Pull request merged**